src/components/clerkcustom/invite-form.tsx

```diff
@@ -0,0 +1,207 @@
+    const [loading, setLoading] = useState(false)
+    const [success, setSuccess] = useState(false)
+    const [formValues, setFormValues] = useState({
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        roles: {
+            reviewer: false,
+            researcher: false,
+        },
+    })
+    const [selectedOrganization, setSelectedOrganization] = useState<string | null>(null)
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault()
+        setError(null)
+        setSuccess(false)
+        setLoading(true)
+
+        const formData = new FormData(e.currentTarget)
+        const firstName = formData.get('firstName')?.toString() ?? ''
+        const lastName = formData.get('lastName')?.toString() ?? ''
+        const email = formData.get('email')?.toString() ?? ''
+        const password = formData.get('password')?.toString() ?? ''
+
+        // Check if organization is selected
+        if (!selectedOrganization) {
+            setError('Please select an organization')
+            setLoading(false)
+            return
+        }
+
+        // Check if at least one role is selected
+        if (!formValues.roles.reviewer && !formValues.roles.researcher) {
+            setError('Please select at least one role (Reviewer or Researcher)')
+            setLoading(false)
+            return
+        }
+
+        // Update form values state
+        setFormValues({
+            firstName,
+            lastName,
+            email,
+            password,
+            roles: formValues.roles, // Keep the existing roles structure
+        })
+
+        try {
+            // save user in Clerk
+            const result = await createClerkUserAction({
+                firstName,
+                lastName,
+                email,
+                password,
+            })
+            // save user in DB
+            await createUserAction(
+                result.clerkId,
+                `${firstName} ${lastName}`,
+                formValues.roles.researcher,
+                selectedOrganization, // Use the selected organization ID directly
+                formValues.roles.reviewer,
+            )
+
+            if (result.success) {
+                // Reset form state
+                setFormValues({
+                    firstName: '',
+                    lastName: '',
+                    email: '',
+                    password: '',
+                    roles: {
+                        reviewer: false,
+                        researcher: false,
+                    },
+                })
+                setSuccess(true)
+            }
+        } catch (error: unknown) {
+            const errorMessage = error instanceof Error ? error.message : 'User creation failed'
+            setError(errorMessage)
+        } finally {
+            setLoading(false)
+        }
+    }
+
+    return (
+        <form onSubmit={handleSubmit}>
```

Comment:
This form can probably be re-written and simplified a ton with [mantine's use-form](https://mantine.dev/form/use-form/) hook and zod for validation, we have a few examples in the codebase using `use-form`. Taking this approach would simplify your `handleSubmit` method considerably as well

i'm happy to jump in and implement the form if you haven't used it before, but lmk! 👍